src/metabase/query_processor/middleware/auto_parse_filter_values.clj

```diff
@@ -0,0 +1,44 @@
+(ns metabase.query-processor.middleware.auto-parse-filter-values
+  "Middleware that parses filter clause values that come in as strings (e.g. from the API) to the appropriate type. E.g.
+  a String value in a filter clause against a `:type/Integer` Field should get parsed into an integer.
+
+  Note that logic for automatically parsing temporal values lives in the `wrap-values-literals` middleware for
+  historic reasons. When time permits it should be moved into this middleware since it's really a separate
+  transformation from wrapping the value literals themselves."
+  (:require [metabase.mbql.util :as mbql.u]
+            [metabase.query-processor.error-type :as error-type]
+            [metabase.util
+             [i18n :refer [tru]]
+             [schema :as su]]
+            [schema.core :as s]))
+
+(s/defn ^:private parse-value-for-base-type [v :- s/Str, base-type :- su/FieldType]
+  {:pre [(string? v)]}
+  (try
+    (condp #(isa? %2 %1) base-type
+      :type/BigInteger (bigint v)
+      :type/Integer    (Integer/parseInt v)
```

Comment:
I don't see `Long` in here, what happens for >32 bit integers?

Reply:
Good point. Fixed

Reply:
without looking, I bet we do this elsewhere

Reply:
did a quick grep of the codebase and couldn't find it anywhere else. We are using `Long/parseLong` in 4 other places tho.